automation/src/main/java/org/wso2/testgrid/automation/reader/JMeterTestReader.java

```diff
@@ -44,19 +47,27 @@
      * @return a list of {@link Test} instances
      */
     private List<Test> processTestStructure(File file, TestScenario scenario) throws TestAutomationException {
+        String preScript = "";
+        String postScript = "";
         List<Test> testsList = new ArrayList<>();
         File tests = new File(file.getAbsolutePath() +
-                              File.separator + JMETER_TEST_PATH);
+                File.separator + JMETER_TEST_PATH);
         List<String> jmxList = new ArrayList<>();
         if (tests.exists()) {
-            for (String jmx : ArrayUtils.nullToEmpty(tests.list())) {
-                if (jmx.endsWith(TestGridConstants.JMTER_SUFFIX)) {
-                    jmxList.add(tests.getAbsolutePath() + File.separator + jmx);
+            for (String script : ArrayUtils.nullToEmpty(tests.list())) {
```

Comment:
There are 2 improvements we can do here,

- Use Java 8 streams to filter the files by extension
- Validate for multiple pre / post scripts since we allow only 1 prescript and a 1 postscript (both are optional)

WDYT?

Reply:
Used streams to filter the scrips.
Initially, there will be only one pre and post script provided by the QA team.